addons/purchase_stock/models/stock_move.py

```diff
@@ -26,6 +26,12 @@ def _prepare_merge_moves_distinct_fields(self):
     def _prepare_merge_negative_moves_excluded_distinct_fields(self):
         return super()._prepare_merge_negative_moves_excluded_distinct_fields() + ['created_purchase_line_id']
 
+    @api.depends('picking_id.partner_id')
+    def _compute_picking_partner_id(self):
+        for move in self:
+            if not move._is_dropshipped() and move in move.picking_id.move_ids_without_package:
+                move.partner_id = move.picking_id.partner_id
+
```

Comment:
The `@depends` for an inherited compute method will auto-inherit, so no need to duplicate them. Also it's typically better to use super() with computes so we don't have to duplicate code + code changes of the original method.

```suggestion
    def _compute_partner_id(self):
        # dropshipped moves should have their partner_ids directly set
        not_dropshipped_moves = self.filtered(lambda m: not m._is_dropshipped())
        super(StockMove, not_dropshipped_moves)._compute_partner_id()

```